api-participant/src/main/java/bio/terra/pearl/api/participant/service/ParticipantTaskExtService.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.pearl.api.participant.service;
+
+import bio.terra.pearl.core.model.EnvironmentName;
+import bio.terra.pearl.core.model.participant.Enrollee;
+import bio.terra.pearl.core.model.participant.ParticipantUser;
+import bio.terra.pearl.core.model.survey.Survey;
+import bio.terra.pearl.core.model.workflow.ParticipantTask;
+import bio.terra.pearl.core.model.workflow.TaskType;
+import bio.terra.pearl.core.service.participant.EnrolleeService;
+import bio.terra.pearl.core.service.portal.PortalWithPortalUser;
+import bio.terra.pearl.core.service.survey.SurveyService;
+import bio.terra.pearl.core.service.workflow.ParticipantTaskService;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ParticipantTaskExtService {
+  private ParticipantTaskService participantTaskService;
+  private SurveyService surveyService;
+  private EnrolleeService enrolleeService;
+  private AuthUtilService authUtilService;
+
+  @Autowired
+  public ParticipantTaskExtService(
+      ParticipantTaskService participantTaskService,
+      SurveyService surveyService,
+      EnrolleeService enrolleeService,
+      AuthUtilService authUtilService) {
+    this.participantTaskService = participantTaskService;
+    this.surveyService = surveyService;
+    this.enrolleeService = enrolleeService;
+    this.authUtilService = authUtilService;
+  }
+
+  // Loads all outreach activities for the logged-in PortalParticipantUser
+  public List<TaskAndSurvey> listSurveyTasks(
+      ParticipantUser user, String portalShortcode, EnvironmentName envName, TaskType taskType) {
+    PortalWithPortalUser portalUser =
+        authUtilService.authParticipantToPortal(user.getId(), portalShortcode, envName);
+    List<Enrollee> participantEnrollees =
+        enrolleeService.findByPortalParticipantUser(portalUser.ppUser());
+
+    List<ParticipantTask> outreachTasks =
+        participantTaskService
+            .findByEnrolleeIds(participantEnrollees.stream().map(Enrollee::getId).toList())
+            .values()
+            .stream()
+            .flatMap(Collection::stream)
+            .filter(task -> taskType == null || task.getTaskType().equals(taskType))
```

Comment:
Maybe I'm misunderstanding, but this would let ParticipantTasks with `taskType==null` through, is that the intended behavior? Or should it be `taskType != null && task.getTaskType().equals(taskType)`?

Reply:
I see your confusion, but the function is correct.  `taskType` is the argument to the API function.  If the client passes `null`, that means they want to see all task types, and so that's why the null short-circuit is here.